tooling/nargo_cli/src/cli/compile_cmd.rs

```diff
@@ -177,20 +194,33 @@ fn compile_program(
         None
     };
 
-    let (program, warnings) =
-        match noirc_driver::compile_main(&mut context, crate_id, compile_options, cached_program) {
-            Ok(program_and_warnings) => program_and_warnings,
-            Err(errors) => {
-                return (context.file_manager, Err(errors));
-            }
-        };
+    // If we want to output the debug information then we need to perform a full recompilation of the ACIR.
+    let force_recompile = output_debug;
```

Comment:
Did you just do this to clarify what a requested debug output will do?

Reply:
Yeah, `output_debug` doesn't really have any meaning inside of `noirc_driver` as it's not responsible for outputting a debug artifact anyway. Because of that I wanted to just make it clear why we're passing in this value.